### ISSUE TITLE
Update for Angular 15

### ISSUE THREAD

**Issue by jasonhodges** [just-jeb/angular-builders#1312]
What is the timeline to update @angular-builders/custom-webpack for Angular 15? Thanks

**Comment by just-jeb** [comment]
Probably a week after Angular team release a stable version.

**Comment by eng-dibo** [comment]
Angular v15 has been officially released @just-jeb

**Comment by eng-dibo** [comment]
please note that Angular is going to use `esbuild` for future builders

**Comment by fasidOnGit** [comment]
Would be great if you could also expose a way to use the experimental `esbuild` on v15. Thanks :)

**Comment by CarlosTorrecillas** [comment]
It would be nice if this can be achieved soon since we can't upgrade to Angular 15 until you release the new compatible version of the package :)

**Comment by vzakharov-rxnt** [comment]
> please note that Angular is going to use esbuild for future builders

@eng-dibo esbuild is still experimental in v15. You have to enable it explicitly in angular.json. Also it doesn't work with `ng serve` (yet).

**Comment by just-jeb** [comment]
@eng-dibo Angular doesn't support extension of `esbuild` builder, so there is nothing we can do about it yet.  
https://github.com/just-jeb/angular-builders/issues/1288

**Comment by rpanadero** [comment]
Waiting for the release that supports Angular 15! Thanks in advance!

**Comment by dolanmiu** [comment]
I get the error:

```
This version of CLI is only compatible with Angular versions ^14.0.0,
but Angular version 15.0.0 was found instead.

Please visit the link below to find instructions on how to update Angular.
https://update.angular.io/
```

I suspect it's because `custom-webpack` needs updating?

**Comment by reda-alaoui** [comment]
@dolanmiu , you can add this to `package.json` as a workaround :

```
"resolutions": {
    "@angular-devkit/architect": "^0.1500.0",
    "@angular-devkit/build-angular": "^15.0.0",
    "@angular-devkit/core": "^15.0.0"
  }
```

**Comment by dolanmiu** [comment]
Thanks @reda-alaoui, but i do not use `yarn`, is there an npm alternative?

**Comment by reda-alaoui** [comment]
@dolanmiu I don't use npm, can't help you :/

**Comment by B1Z1** [comment]
Waiting for Angular 15, it would be cool to implement support for it

**Comment by davidpanic** [comment]
We're waiting on this because we use the Sentry webpack plugin which requires this.

**Comment by muuvmuuv** [comment]
Guys, please stop spamming. Angular 15 is just a week old, you don't need to update all your dependencies every time a new release comes out. Your software worked until today, so will tomorrow.

**Comment by fklMi** [comment]
@B1Z1 You can add significance to an issue by upvoting or subscribing to it

**Comment by just-jeb** [comment]
Guys, be kind to each other 😄 We want it as much as you do and we work on it, @arturovt has already put an initial PR (#1319), still have a few things to fix. It won't take long, but please be patient.

**Comment by arturovt** [comment]
> @muuvmuuv it's not about spam. Just a reminder that the feature is important and people waiting for it. No one hating here

It is **about spam**. Repeatedly leaving comments in this issue for us to publish faster is not helpful or necessary. Please do not repeatedly mention us if we don't reply as fast as you'd like us to.

We **are working** on this in pairs daily with just-jeb and resolving upcoming issues to release a v15 version ASAP. Your organizations can use the modified version through _yarn-resolutions_ or _npm-overrides_ for now before the v15 is released.

**Comment by B1Z1** [comment]
> > @muuvmuuv it's not about spam. Just a reminder that the feature is important and people waiting for it. No one hating here
> 
> It is **about spam**. Repeatedly leaving comments in this issue for us to publish faster is not helpful or necessary. Please do not repeatedly mention us if we don't reply as fast as you'd like us to.
> 
> We **are working** on this in pairs daily with just-jeb and resolving upcoming issues to release a v15 version ASAP. Your organizations can use the modified version through _yarn-resolutions_ or _npm-overrides_ for now before the v15 is released.

Ok, I'm sorry

**Comment by just-jeb** [comment]
Beta version released, you can install it using `next` tag.  
Huge shoutout to @arturovt who took it upon himself and made it happen so fast 🚀! 

The stable version will be released after we update Jest builder dependencies (which is blocked by `jest-preset-angular` to update to `ts-jest` 29). Since it's going to include potential breaking changes we can't release a stable version before.

**Comment by vzakharov-rxnt** [comment]
@just-jeb We are using karma. Is it still supported?

**Comment by just-jeb** [comment]
It is.

**Comment by ItsMeSousuke** [comment]
> Beta version released, you can install it using `next` tag. Huge shoutout to @arturovt who took it upon himself and made it happen so fast 🚀!
> 
> The stable version will be released after we update Jest builder dependencies (which is blocked by `jest-preset-angular` to update to `ts-jest` 29). Since it's going to include potential breaking changes we can't release a stable version before.

`jest-preset-angular` was just released :)

**Comment by Anlanther** [comment]
> Error: Schema validation failed with the following errors: Data path "/polyfills" must be string

I noticed that for Angular v15 `angular.json`, polyfills is now an array rather than just a string

**Comment by just-jeb** [comment]
@ItsMeSousuke it was, but if you look closely at the release it doesn't contain any changes except for updated `peerDependencies`. While we will release a new beta version containing this change for Jest builder, we'll still have to wait for the potentially breaking change (which is the [upgrade to Jest 29](https://github.com/thymikee/jest-preset-angular/pull/1840#issuecomment-1325671764)) in order to release a stable version.

**Comment by lukaszgodula** [comment]
@just-jeb PR you mentioned is already closed, do you think it is now possible to go with stable version?

**Comment by just-jeb** [comment]
The PR supports Angular 15 but not Jest 29, which is the potential breaking change. But since I don't see any advancement on this matter from their side I'm considering releasing the stable version for Angular 15 and postpone the Jest 29 upgrade to version 16 (hopefully by then `jest-preset-angular` will support it). 
@arturovt WDYT? Should we go ahead and graduate v15?

**Comment by dolanmiu** [comment]
Sounds good @just-jeb. Can always create a minor or patch release to support Jest 29 if needs be

> potential breaking change

What could those potential breaking changes look like? Not heard anyone complain about it, so perhaps its fine?

**Comment by SebastianStehle** [comment]
A release would be awesome

**Comment by SebastianStehle** [comment]
I just tested the beta. On the build server I get the following error (what I do not understand):

```
Error: Could not find the '@angular-builders/custom-webpack:karma' builder's node package.
```

Locally (Windows) I get the following:

```
Chrome Headless 108.0.5359.125 (Windows 10) ERROR                                                                                                                                                                                     
  An error was thrown in afterAll
  Uncaught TypeError: __webpack_require__(...).context is not a function
  TypeError: __webpack_require__(...).context is not a function
      at Module.4289 (http://localhost:9876/_karma_webpack_/webpack:/src/test.ts:24:25)
      at __webpack_require__ (http://localhost:9876/_karma_webpack_/webpack:/webpack/bootstrap:19:1)
      at __webpack_exec__ (http://localhost:9876/_karma_webpack_/main.js:394332:48)
      at http://localhost:9876/_karma_webpack_/main.js:394333:54
      at Function.__webpack_require__.O (http://localhost:9876/_karma_webpack_/webpack:/webpack/runtime/chunk loaded:23:1)
      at http://localhost:9876/_karma_webpack_/main.js:394334:56
      at webpackJsonpCallback (http://localhost:9876/_karma_webpack_/webpack:/webpack/runtime/jsonp chunk loading:34:1)
      at http://localhost:9876/_karma_webpack_/main.js:1:67
```

**Comment by just-jeb** [comment]
@dolanmiu I'm talking about [this](https://github.com/kulshekhar/ts-jest/blob/main/CHANGELOG.md#deprecations). Since we're essentially merging user configs with a predefined one, the change on our side would most likely require a change on user side (if they have custom config). Hence I'm saying breaking change (which cannot be released as minor/patch version). 

@SebastianStehle looks like dependencies issue. Try removing `package-lock.json`/`yarn.lock` and `node_modules` and running a clean install.

**Comment by SebastianStehle** [comment]
Forget what I said. There are 2 issues:

1. I changed my base image, which sets NODE_ENV = production
2. I still had some old config in my angular project. Not sure why it was not migrated, but I compared it with a new project and fixed that.

**Comment by just-jeb** [comment]
Angular Builders v15 is out. 

Jest 29 will be officially supported in v16, however, if your setup is straight forward it will probably work with v15 (the naive [PR](https://github.com/just-jeb/angular-builders/pull/1309) with all the deps updated to Jest 29 passes).
